docs/tutorials/setup-env.md

```diff
@@ -0,0 +1,165 @@
+---
+id: setup-env
+title: Environment Setup 
+---
+This tutorial will help you set up your local development environment to build your first DHIS2 app using the [DHIS2 Application Platform](https://platform.dhis2.nu/#/), which provides a fully-featured DHIS2 app, with out-of-the-box tools and components ready to be customized for your use case. 
+
+In this section we will:
+
+1. Install the [DHIS2 CLI](https://cli.dhis2.nu/#/)
+2. Spin up a DHIS2 instance with [d2 cluster](https://cli.dhis2.nu/#/commands/d2-cluster) 
+3. Initialize a new DHIS2 app 
+
+## Before you start 
+
+You will need to have the following installed before proceeding:
+
+1. Install [Node.js](https://nodejs.org/en/download/)
+
+Make sure that you are running a compatible version of Node as using older versions may throw some errors during setup. 
+
+2. Install [Yarn](https://yarnpkg.com/getting-started/install)
+
+## 1. Install the DHIS2 CLI
+
+We'll start by installing the [DHIS2 CLI](https://cli.dhis2.nu/#/) (Command Line Interface) globally. 
+
+The `@dhis2/cli` package provides a set of tools that are useful when developing DHIS2 apps. It contains a number of modules that lets you quickly create and maintain DHIS2 applications. It also allows you to manage local DHIS2 instances by running simple commands. 
+
+To install `@dhis2/cli` globally run the following command using `Yarn`:  
+
+```shell
+yarn global add @dhis2/cli
+```
+
+Verify your installation and check all the commands that are available to you: 
+
+```shell
+d2 --help
+```
+
+You’re now ready to start using `@dhis2/cli` commands to spin up a DHIS2 local instance and to create a new app! ✨
+
+## 2. Spin up a DHIS2 instance with `d2 cluster`
+
+_This is a very important step as you will need to connect your new app to this DHIS2 instance in [Step 3](#3-initialize-a-new-dhis2-app), which will be your **server**._  
+
+The [d2 cluster](https://cli.dhis2.nu/#/commands/d2-cluster) command will allow you to launch a new cluster and create a DHIS2 instance using Docker containers. 
+
+Some of the benefits of using local instances is that we can run multiple different DHIS2 versions at the same time and also makes testing easier as it runs locally (and it's not dependent on a network connection). 
+
+Before you start, you would need to have Docker installed on your computer. So please go ahead and download [Docker](https://www.docker.com/).  
+
+Now you can quickly start a new DHIS2 instance by running the following command from your terminal: 
+
+```shell
+d2 cluster up <name>
+# or:
+d2 cluster up 2.35.0 --db-version 2.35 —-seed
+# this will populate a database with sample data
```

Comment:
Once the `up` command completes, the server might take up to a couple minutes to start.  It could be confusing if the server isn't available right away.  i usually recommend running `d2 cluster logs 2.35.0` and waiting until you see the line "server startup in xxx ms" line.  We should also add an auto-wait mechanism to the `up` command and/or an informative 503 error page to nginx (I digress...)